include/exiv2/matroskavideo.hpp

```diff
@@ -0,0 +1,141 @@
+
+// *****************************************************************************
+// class definitions
+namespace Internal {
+
+/// @brief  Utility function to search into std::array of pairs
+/// @return the searched pair if exist,else nullptr
+template <size_t N>
+[[nodiscard]] const Exiv2::MatroskaTags* findTag(const std::array<Exiv2::MatroskaTags, N>& src, const uint64_t& key) {
+  const auto rc =
+      std::find_if(src.begin(), src.end(), [&key](const Exiv2::MatroskaTags& element) { return element.first == key; });
+  // the return value is of type "const Exiv2::MatroskaTags*", so we return the adress of the content of the input
+  // iterator return by find_if
+  return rc == std::end(src) ? nullptr : &(*rc);
+}
+}  // namespace Internal
+
+/*!
+  @brief Class to access Matroska video files.
+ */
+class EXIV2API MatroskaVideo : public Image {
+ public:
+  //! @name Creators
+  //@{
+  /*!
+    @brief Constructor for a Matroska video. Since the constructor
+        can not return a result, callers should check the good() method
+        after object construction to determine success or failure.
+    @param io An auto-pointer that owns a BasicIo instance used for
+        reading and writing image metadata. \b Important: The constructor
+        takes ownership of the passed in BasicIo instance through the
+        auto-pointer. Callers should not continue to use the BasicIo
+        instance after it is passed to this method. Use the Image::io()
+        method to get a temporary reference.
+   */
+  explicit MatroskaVideo(BasicIo::UniquePtr io);
+
+  //! Copy constructor
+  MatroskaVideo(const MatroskaVideo& rhs) = delete;
+  //! Assignment operator
+  MatroskaVideo& operator=(const MatroskaVideo& rhs) = delete;
+  //@}
+
+  //! @name Manipulators
+  //@{
+  void readMetadata() override;
+  void writeMetadata() override;
+  //@}
+
+  //! @name Accessors
+  //@{
+  [[nodiscard]] std::string mimeType() const override;
+  //@}
+
+ protected:
+  /*!
+    @brief Function used to calulate the size of a block.
+        This information is only stored in one byte.
+        The size of the block is calculated by counting
+        the number of leading zeros in the binary code of the byte.
+        Size = (No. of leading zeros + 1) bytes
+    @param b The byte, which stores the information to calculate the size
+    @return Return the size of the block.
+   */
+  [[nodiscard]] uint32_t findBlockSize(byte b);
+  /*!
+    @brief Check for a valid tag and decode the block at the current IO position.
+        Calls contentManagement() or skips to next tag, if required.
+   */
+  void decodeBlock();
+  /*!
+    @brief Interpret tag information, and save it in the respective XMP container.
+    @param mt Pointer to current tag,
+    @param buf Pointer to the memory area with the tag information.
+    @param size Size of \em buf.
+   */
+  void contentManagement(const MatroskaTags* mt, const byte* buf, long size);
```

Comment:
Please don't use the `long` type. We're trying to eliminate it from the codebase. We usually use `size_t` for sizes.

Reply:
understood, thank you very much @kevinbackhouse  for your time.
I will analyse the crash report and run locally the fuzz test to fix them

Reply:
replacing long by size_t done here [f14bf95](https://github.com/Exiv2/exiv2/pull/2370/commits/f14bf95862ddaf4b02fafb5221b97319b4ef4e81)